src/components/text-editor/text-editor.pw.tsx

```diff
@@ -0,0 +1,613 @@
+    });
+
+    await textEditorInput(page).focus();
+    await expect(textEditorContainer(page).locator("..")).toHaveCSS(
+      "outline",
+      "rgb(255, 188, 25) solid 3px"
+    );
+
+    await textEditorToolbar(page, "bold").focus();
+    await checkGoldenOutline(textEditorToolbar(page, "bold"), "2px");
+
+    await textEditorToolbar(page, "italic").focus();
+    await checkGoldenOutline(textEditorToolbar(page, "italic"), "2px");
+
+    await textEditorToolbar(page, "bullet-list").focus();
+    await checkGoldenOutline(textEditorToolbar(page, "bullet-list"), "2px");
+
+    await textEditorToolbar(page, "number-list").focus();
+    await checkGoldenOutline(textEditorToolbar(page, "number-list"), "2px");
+  });
+
+  test(`should render with the expected styling when focusRedesignOptOut is false`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<TextEditorCustom />);
+
+    const editorParent = textEditorContainer(page).locator("..");
+    const toolbarBold = textEditorToolbar(page, "bold");
+    const toolbarItalic = textEditorToolbar(page, "italic");
+    const toolbarBullet = textEditorToolbar(page, "bullet-list");
+    const toolbarNumber = textEditorToolbar(page, "number-list");
+
+    await textEditorInput(page).focus();
+    await expect(editorParent).toHaveCSS(
+      "box-shadow",
+      "rgb(255, 188, 25) 0px 0px 0px 3px, rgba(0, 0, 0, 0.9) 0px 0px 0px 6px"
+    );
+    await expect(editorParent).toHaveCSS(
+      "outline",
+      "rgba(0, 0, 0, 0) solid 3px"
+    );
+
+    await toolbarBold.focus();
+    await expect(toolbarBold).toHaveCSS(
+      "box-shadow",
+      "rgb(255, 188, 25) 0px 0px 0px 3px, rgba(0, 0, 0, 0.9) 0px 0px 0px 6px"
+    );
+    await expect(toolbarBold).toHaveCSS(
+      "outline",
+      "rgba(0, 0, 0, 0) solid 3px"
+    );
+    await expect(toolbarBold).toHaveCSS("position", "relative");
+
+    await toolbarItalic.focus();
+    await expect(toolbarItalic).toHaveCSS(
+      "box-shadow",
+      "rgb(255, 188, 25) 0px 0px 0px 3px, rgba(0, 0, 0, 0.9) 0px 0px 0px 6px"
+    );
+    await expect(toolbarItalic).toHaveCSS(
+      "outline",
+      "rgba(0, 0, 0, 0) solid 3px"
+    );
+    await expect(toolbarItalic).toHaveCSS("position", "relative");
+
+    await toolbarBullet.focus();
+    await expect(toolbarBullet).toHaveCSS(
+      "box-shadow",
+      "rgb(255, 188, 25) 0px 0px 0px 3px, rgba(0, 0, 0, 0.9) 0px 0px 0px 6px"
+    );
+    await expect(toolbarBullet).toHaveCSS(
+      "outline",
+      "rgba(0, 0, 0, 0) solid 3px"
+    );
+    await expect(toolbarBullet).toHaveCSS("position", "relative");
+
+    await toolbarNumber.focus();
+    await expect(toolbarNumber).toHaveCSS(
+      "box-shadow",
+      "rgb(255, 188, 25) 0px 0px 0px 3px, rgba(0, 0, 0, 0.9) 0px 0px 0px 6px"
+    );
+    await expect(toolbarNumber).toHaveCSS(
+      "outline",
+      "rgba(0, 0, 0, 0) solid 3px"
+    );
+    await expect(toolbarNumber).toHaveCSS("position", "relative");
+  });
+});
+
```

Comment:
We should have tests here for the optional buttons too (e.g. Save and Cancel). Mainly for the operation using the keyboard and focus states as Chromatic can snapshot the default state. When they're present, the right arrow key should cycle back to the styling keys but pressing Tab on the numbered-list key should move focus to the first optional Button.

Reply:
We do already have this unit tested so I would be inclined for now to leave it as is until we can do a full review of what we test where when this playwright epic is finished.